packages/next/src/build/webpack/plugins/next-types-plugin/index.ts

```diff
@@ -158,14 +159,14 @@ if ('generateViewport' in entry) {
 }
 // Check the arguments and return type of the generateStaticParams function
 if ('generateStaticParams' in entry) {
-  checkFields<Diff<{ params: PageParams }, FirstArg<MaybeField<TEntry, 'generateStaticParams'>>, 'generateStaticParams'>>()
+  checkFields<Diff<{ params: SegmentParams }, FirstArg<MaybeField<TEntry, 'generateStaticParams'>>, 'generateStaticParams'>>()
   checkFields<Diff<{ __tag__: 'generateStaticParams', __return_type__: any[] | Promise<any[]> }, { __tag__: 'generateStaticParams', __return_type__: ReturnType<MaybeField<TEntry, 'generateStaticParams'>> }>>()
 }
 
-type PageParams = any
+type SegmentParams = {[param: string]: string | string[] | undefined}
```

Comment:
This will error because params is now a promise. but this will definitely work

```
type PageProps = {
   params: Promise<{ abc: string | undefined }>
}

function Page(props: PageProps) {}
```